js/common/lib/training-session-impl.ts

```diff
@@ -176,12 +176,22 @@ export class TrainingSession implements TrainingSessionInterface {
     return this.convertHandlerReturnTypeToMapOfTensors(results);
   }
 
-  async loadParametersBuffer(_array: Uint8Array, _trainableOnly: boolean): Promise<void> {
-    throw new Error('Method not implemented.');
+  async getParametersSize(trainableOnly: boolean): Promise<number> {
+    return this.handler.getParametersSize(trainableOnly);
   }
 
-  async getContiguousParameters(_trainableOnly: boolean): Promise<Uint8Array> {
-    throw new Error('Method not implemented.');
+  async loadParametersBuffer(array: Float32Array, trainableOnly: boolean): Promise<void> {
+    const paramsSize = await this.getParametersSize(trainableOnly);
+    if (array.length !== paramsSize) {
+      throw new Error(
+          'Size of the buffer passed into loadParametersBuffer must match the number of parameters in ' +
+          'the model. Please use getParametersSize method to check.');
+    }
+    return this.handler.loadParametersBuffer(array, trainableOnly);
+  }
+
+  async getContiguousParameters(trainableOnly: boolean): Promise<OnnxValue> {
+    return this.handler.getContiguousParameters(trainableOnly);
```

Comment:
user does not have an option to pass a preallocated buffer? Is this a limitation from CAPI or we decided on the js side?

Reply:
The user would not be able to pre-allocate a buffer on the WASM side since those functions are obscured from the user, as far as I know -- we can keep it in mind as something to investigate in future releases. The C API takes in a pre-allocated buffer as one of its parameters.